include/dynd/types/type_id.hpp

```diff
@@ -355,6 +355,11 @@ struct type_id_of<void> {
   static const type_id_t value = void_id;
 };
 
+template <>
+struct type_id_of<ndt::type> {
+  static const type_id_t value = type_id;
+};
+
```

Comment:
This one could also go into the ```property_type_id_of``` set.  I wasn't sure if it was deliberately missing here or not.